azure-devops/azext_devops/dev/repos/ref.py

```diff
@@ -0,0 +1,131 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from knack.util import CLIError
+from vsts.git.v4_0.models.git_ref_update import GitRefUpdate
+from vsts.exceptions import VstsServiceError
+from azext_devops.dev.common.git import resolve_git_refs
+from azext_devops.dev.common.services import (get_git_client,
+                                              resolve_instance_project_and_repo)
+
+logger = get_logger(__name__)
+
+
+# pylint: disable=redefined-builtin
+def list_refs(filter=None, repository=None, organization=None, project=None, detect=None):
+    """List the references.
+    :param str filter: A filter to apply to the refs (starts with). Example: head or heads/ for the branches.
+    :param str repository: Name or ID of the repository.
+    :param str organization: Azure Devops organization URL. Example: https://dev.azure.com/MyOrganizationName/
+    :param str project: Name or ID of the project.
+    :param str detect: Automatically detect organization and project. Default is "on".
+    """
+    try:
+        organization, project, repository = resolve_instance_project_and_repo(
+            detect=detect,
+            organization=organization,
+            project=project,
+            repo=repository)
+        client = get_git_client(organization)
+        return client.get_refs(repository_id=repository,
+                               project=project,
+                               filter=filter)
+    except VstsServiceError as ex:
+        raise CLIError(ex)
+
+
+def create_ref(name, object_id, repository=None, organization=None, project=None, detect=None):
+    """Create a reference.
+    :param str name: Name of the reference to create (example: heads/my_branch or tags/my_tag).
+    :param str object_id: Id of the object to create the reference from.
+    :param str repository: Name or ID of the repository.
+    :param str organization: Azure Devops organization URL. Example: https://dev.azure.com/MyOrganizationName/
+    :param str project: Name or ID of the project.
+    :param str detect: Automatically detect organization and project. Default is "on".
+    """
+    try:
+        organization, project, repository = resolve_instance_project_and_repo(
+            detect=detect,
+            organization=organization,
+            project=project,
+            repo=repository)
+        client = get_git_client(organization)
+        # by default, the create method does not support setting the is_locked value
+        # to True.
+        ref_update = GitRefUpdate(is_locked=False,
+                                  name=resolve_git_refs(name),
```

Comment:
Why are we not using [resolve_git_ref_heads](https://github.com/Microsoft/azure-devops-cli-extension/blob/master/azure-devops/azext_devops/dev/common/git.py#L115)